### PR TITLE
filter design "prototypes"

### PR DESCRIPTION
**Pull request by ev-br**
- [x] `buttap`
- [x] `cheb1ap`
- [x] `cheb2ap`
- [x] `ellipap` --- requires elliptic integrals, so add them to `cupyx.scipy.special`
- [ ] `besselap` -- will do in a follow-up PR

cupyx/scipy/special
- [x] `ellipk`
- [x] `ellipkm1`
- [x] `ellipj`

### CODE REVIEW

**Comment by ev-br** [comment]
Would appreciate help with the CI failures. AFAIU, the issue is that this PR adds a couple of ufuncs in scipy.special with `.types = "d->d"` only, while their scipy.special analogs also have an `f->f` loop. Do I need to add a special case to https://github.com/cupy/cupy/blob/main/tests/cupyx_tests/scipy_tests/special_tests/test_ufunc_dispatch.py#L44 ?

**Comment by takagi** [comment]
LGTM except for the static checks!

**Comment by ev-br** [comment]
A rebase on main fixed the style checks, it seems.

**Comment by takagi** [comment]
doc build does not have CUDA, so `_POW10_LOG10 = cupy.log(10)` causes an error on import time. Having numpy scalar, `numpy.log(10)`, would be enough for the precomputed value to multiply with a `cupy.ndarray` instead.

**Comment by ev-br** [comment]
Thanks! using `math.log(10)` seems to have done the trick.

**Comment by takagi** [comment]
> Thanks! using math.log(10) seems to have done the trick.

Sound good! FYI, scalar values are passed to GPU via cudaLaunchKernel's argument, so we don't need explicit H2D copy here.

**Review by takagi — APPROVED** [review]
LGTM!

File: tests/cupyx_tests/scipy_tests/signal_tests/test_iir_filter_conversion.py [hunk 1]
```diff
@@ -237,3 +237,22 @@ def test_basic(self, xp, scp):
...
+
+    @testing.numpy_cupy_allclose(scipy_name="scp")
+    def test_cheb1ap(self, xp, scp):
+        return scp.signal.cheb1ap(3, 1)
+
+    @testing.numpy_cupy_allclose(scipy_name="scp")
+    def test_cheb2ap(self, xp, scp):
+        return scp.signal.cheb2ap(3, 1)
+
+    @testing.numpy_cupy_allclose(scipy_name="scp", atol=1e-5, rtol=1e-5)
```
Comment (takagi) [inline tests/cupyx_tests/scipy_tests/signal_tests/test_iir_filter_conversion.py:256]:
```suggestion
    @testing.numpy_cupy_allclose(scipy_name="scp", atol=3e-5, rtol=3e-5)
```

**Comment by takagi** [comment]
Thanks, @ev-br!

**Pull request merged**